kv/kvserver: TestReplicateRemovedNodeDisruptiveElection failed

[(kv/kvserver).TestReplicateRemovedNodeDisruptiveElection failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2745063&tab=buildLog) on [master@7ddc401e9f7016df9057abf01a9fb5825e8f5720](https://github.com/cockroachdb/cockroach/commits/7ddc401e9f7016df9057abf01a9fb5825e8f5720):

```
W210304 17:01:13.144318 414784 sql/sqlliveness/slinstance/slinstance.go:183  [n2] 15  exiting heartbeat loop
    panic.go:969: runtime error: invalid memory address or nil pointer dereference
        goroutine 413292 [running]:
        runtime/debug.Stack(0xc0077d2d48, 0x41186c0, 0x72e29a0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:110 +0x278
        panic(0x41186c0, 0x72e29a0)
        	/usr/local/go/src/runtime/panic.go:969 +0x1b9
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Stop(0xc004800200, 0x54d1900, 0xc00019e010)
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:470 +0x31e
        panic(0x41186c0, 0x72e29a0)
        	/usr/local/go/src/runtime/panic.go:969 +0x1b9
        github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).GetRaftLeader.func1.1(0xc000259c00, 0x42b985, 0xc0077d3238)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1397 +0x7d
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Stores).VisitStores.func1(0x1, 0xc000259c00, 0xc0077d3238)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/stores.go:131 +0x38
        github.com/cockroachdb/cockroach/pkg/util/syncutil.(*IntMap).Range(0xc006ef17d0, 0xc0077d32c8)
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/syncutil/int_map.go:352 +0x130
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Stores).VisitStores(0xc006ef17a0, 0xc0077d3370, 0xc000000021, 0x428c10)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/stores.go:130 +0x75
        github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).GetRaftLeader.func1(0x60411259, 0xc0077d33e0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1389 +0x137
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoonError.func1(0x83b68a6, 0xed7d30959)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:47 +0x32
        github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration(0xa7a358200, 0xc0077d34c8, 0x0, 0x6043fa0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:197 +0xe7
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoonError(0xc0077d3588, 0xc0077d3558, 0x5465df)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:53 +0x8c
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x561c900, 0xc007898780, 0xc0077d3588)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:34 +0x45
        github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).GetRaftLeader(0xc003f16480, 0x561c900, 0xc007898780, 0xc00e31cac8, 0x1, 0x1, 0x14de594)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1386 +0xb6
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateRemovedNodeDisruptiveElection.func1(0x60411259, 0xc0077d3760)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_raft_test.go:3464 +0xe9
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoonError.func1(0x83b4dd0, 0xed7d30959)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:47 +0x32
        github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration(0xa7a358200, 0xc0077d3848, 0x0, 0x6043fa0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:197 +0xe7
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoonError(0xc0077d3db0, 0xc007898780, 0xc0077d3898)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:53 +0x8c
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x561c900, 0xc007898780, 0xc0077d3db0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:34 +0x45
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateRemovedNodeDisruptiveElection(0xc007898780)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_raft_test.go:3462 +0x6aa
        testing.tRunner(0xc007898780, 0x4e28bf8)
        	/usr/local/go/src/testing/testing.go:1123 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1168 +0x2b3
--- FAIL: TestReplicateRemovedNodeDisruptiveElection (1.41s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicateRemovedNodeDisruptiveElection PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRemovedNodeDisruptiveElection.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/testutils/testcluster/testcluster.go
+++ b/pkg/testutils/testcluster/testcluster.go
@@ -1394,6 +1394,9 @@ func (tc *TestCluster) GetRaftLeader(t testing.TB, key roachpb.RKey) *kvserver.R
 					return nil
 				}
 				raftStatus := repl.RaftStatus()
+				if raftStatus == nil {
+					return errors.Errorf("raft group is not initialized for replica with key %s", key)
+				}
 				if raftStatus.Term > latestTerm || (raftLeaderRepl == nil && raftStatus.Term == latestTerm) {
 					// If we find any newer term, it means any previous election is
 					// invalid.

